x/investors/keeper/send_restriction.go

```diff
@@ -0,0 +1,58 @@
+package keeper
+
+import (
+	"context"
+
+	sdk "github.com/cosmos/cosmos-sdk/types"
+	distrtypes "github.com/cosmos/cosmos-sdk/x/distribution/types"
+
+	"github.com/milkyway-labs/milkyway/v10/x/investors/types"
+)
+
+// SendRestrictionFn function that can be used in the x/bank module to intercept
+// the transfer from the distribution module account to the delegator address and
+// redirect the rewards to the investors module account. These redirected rewards
+// are then redistributed to the investors in the end blocker, after deducting
+// the amount based on the InvestorsRewardRatio parameter.
+func (k *Keeper) SendRestrictionFn(ctx context.Context, from sdk.AccAddress, to sdk.AccAddress, amount sdk.Coins) (sdk.AccAddress, error) {
```

Comment:
This restriction can be bypassed allowing investor to get all the rewards.
Lets have `inv1` and `account1`, both are controlled by an investor and have withdraw address euqal to their address.
To bypass this logic `account1` can set its withdraw address to `inv1` then `GetDelegatorAddressByWithdrawAddress` will return `account1` which is not an investor and so the rewards will be sent to `inv1` without sending the investor rewards ratio to the comunity pool.
To solve this `GetDelegatorAddressByWithdrawAddress` should return a list of addresses since there can be multiple addresses associated to a single withdraw address.